contracts/kernel/IKernel.sol

```diff
@@ -15,6 +17,8 @@ interface IKernelEvents {
 
 // This should be an interface, but interfaces can't inherit yet :(
 contract IKernel is IKernelEvents, IVaultRecoverable {
+    function setAgreement(IDisputable _disputableApp, IAgreement _agreement) external;
+
     function acl() public view returns (IACL);
     function hasPermission(address who, address where, bytes32 what, bytes how) public view returns (bool);
 
```

Comment:
Two comments:

1. Are we able to move all the interfaces to be external?
2. Should we use `AragonApp` as the type for `setApp()`'s `app` parameter? I'm debating if this interface dependency tree is worth it, but if not, then we may also want to type `setAgreement()` with just addresses.

Reply:
I was planning to discuss this as one of the things to change in the new breaking version, but yea, I'd love to polish all the interfaces we provide